### ISSUE TITLE
aria-role should only be applied to DOM elements

### ISSUE THREAD

**Issue by wereHamster** [evcohen/eslint-plugin-jsx-a11y#166]
I have some data that is fetched from the server, and it comes as an object with these properties: `{ author, role, photo }`. `role` is the person's role inside the company (like `"Chief Something Officer"`). I can't create a component which takes exactly these properties, because of the `jsx-a11y/aria-role` rule. The rule should be changed to only apply to DOM elements and not React components.

**Comment by ljharb** [comment]
Generally, it's best to avoid using DOM attribute names anywhere else, to prevent confusion.

However, I'd be in favor of adding a boolean `ignoreNonDOM` that defaulted to `false`, to allow for this.

**Comment by wereHamster** [comment]
Maybe. But it wasn't my decision to design the API that way. I'm just a consumer.

Of course I could convert `{author,role,photo}` to `{author,authorRole,photo}` and then pass the later as props to the component. Or call the component as a plain function instead of through JSX (ie. instead of `<Author role=".." >` use `{Author({ role: ".." })}`. But both are workarounds, they don't solve the real issue, which is that jsx-a11y doesn't distinguish between DOM props and custom component props.

**Comment by evcohen** [comment]
i think this makes sense. we've gone with the pattern of ignoring non-DOM elements by default in other places, but that is usually when we need information about the DOM element in order to accurately enforce a rule. config option seems like a fair compromise to me!

**Comment by martpie** [comment]
Is this something that changed recently? I am getting "Elements with ARIA roles must use a valid, non-abstract ARIA role." errors on custom components.
